prestodb/redirect.py

```diff
@@ -0,0 +1,34 @@
+import abc
+from future.standard_library import install_aliases
+install_aliases()
+
+from future.utils import with_metaclass
+import ipaddress
+import socket
+from typing import Any  # NOQA
+from urllib.parse import urlparse
+
+
+class RedirectHandler(with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
+    def handle(self, url):
+        pass
+
+
+
+def _normalize_url_with_hostname(url):
+    parsed = urlparse(url.encode('utf-8'))
```

Comment:
Let's add a TODO here; `urllib.urlunparse` is bogus dealing with utf8; there may have a potential problem with `urlparse` too; replace the library in the future if possible.